### PR TITLE
Resolve #1656, #1657, #1664, #1669

### LINKED ISSUE #1656: receiveForm update depth exceeded on edge case
This happens when receive has exactly one matched account, so the auto-selection is executed, casuing repeated form updates.

Need to add a new test case for this situation

### PR DESCRIPTION
**Pull request by timothywangdev**

### CODE REVIEW

File: src/components/AccountDropdownComponent.jsx [hunk 1]
```diff
@@ -222,6 +219,7 @@ class AccountDropdownComponent extends Component<Props, State> {
   }
 
   render () {
+    console.log('render')
```
Comment (InVincible2016) [inline src/components/AccountDropdownComponent.jsx:222]: remove

File: src/components/AccountDropdownComponent.jsx [hunk 2]
```diff
@@ -314,7 +312,7 @@ class AccountDropdownComponent extends Component<Props, State> {
             <MenuItem value='addCryptoAccounts'>
               <Button onClick={() => addAccount()} color='primary' fullWidth>
                 <AddIcon fontSize='small' />
-                  Add connection to your wallet
+                Add connection to your wallet
```
Comment (InVincible2016) [inline src/components/AccountDropdownComponent.jsx:315]: why remove space

Comment (timothywangdev) [reply src/components/AccountDropdownComponent.jsx:315]: this is correct indentation

File: src/components/AccountDropdownComponent.jsx (hunk 1 above)
Comment (timothywangdev) [reply src/components/AccountDropdownComponent.jsx:222]: fixed
